testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -5474,6 +5474,29 @@ private static void checkIf(SqlOperatorFixture f) {
     f.checkNull("array_length(null)");
   }
 
+  @Test void testArrayToStringFunc() {
+    final SqlOperatorFixture f0 = fixture();
+    f0.setFor(SqlLibraryOperators.ARRAY_TO_STRING);
+    f0.checkFails("^array_to_string(array['aa', 'b', 'c'], '-')^", "No match found for function"
+        + " signature ARRAY_TO_STRING\\(<CHAR\\(2\\) ARRAY>, <CHARACTER>\\)", false);
+
+    final SqlOperatorFixture f = f0.withLibrary(SqlLibrary.BIG_QUERY);
+    f.checkScalar("array_to_string(array['aa', 'b', 'c'], '-')", "aa-b-c",
+        "VARCHAR NOT NULL");
+    f.checkScalar("array_to_string(array[null, 'aa', null, 'b', null], '-', 'empty')",
+        "empty-aa-empty-b-empty", "VARCHAR NOT NULL");
+    f.checkScalar("array_to_string(array[null, 'aa', null, 'b', null], '-')", "aa-b",
+        "VARCHAR NOT NULL");
+    f.checkScalar("array_to_string(array[null, x'aa', null, x'bb', null], '-')", "aa-bb",
+        "VARCHAR NOT NULL");
+    f.checkScalar("array_to_string(array['', 'b'], '-')", "-b", "VARCHAR NOT NULL");
+    f.checkScalar("array_to_string(array['', ''], '-')", "-", "VARCHAR NOT NULL");
+    f.checkNull("array_to_string(null, '-')");
+    f.checkNull("array_to_string(array['a', 'b', null], null)");
+    f.checkFails("array_to_string(array[1, 2, 3], '-')",
+        "arrayToString supports only String or ByteString, but got java.lang.Integer", true);
+  }
```

Comment:
sorry for late review, it may not be right here "arrayToString supports only String or ByteString, but got java.lang.Integer", it should throw exception in validate type process, instead of runtime process

Reply:
good idea

Reply:
@liuyongvs  what do you think about the newer version (including the validation check)?

Reply:
@rubenada i have responsed